app/lib/app/SecvisogramPage/View/FormEditor/editors/GenericEditor/Attributes/CVSS2Attribute.js

```diff
@@ -0,0 +1,211 @@
+import TextAttribute from './TextAttribute.js'
+import Attribute from './shared/Attribute.js'
+import DocumentEditorContext from '../../../../shared/DocumentEditorContext.js'
+import { cvssDropdown, getSeverityColors } from './shared/cvssUtils.js'
+import {
+  vectorUpdateBaseScore,
+  vectorUpdateEnvironmentalScore,
+  vectorUpdateFromVectorString,
+  vectorUpdateTemporalScore,
+  vectorUpdateVectorString,
+} from '../../../../../../../shared/cvss2Tools.js'
+import { set } from 'lodash/fp.js'
+
+/**
+ * @param {{
+ *  instancePath: string[]
+ *  value: {[key: string]: string | number }
+ *  property: import('../../../shared/types').Property
+ *  disabled: boolean
+ * }} props
+ */
+export default function CVSSV2Attribute({
+  instancePath,
+  value,
+  property,
+  disabled,
+}) {
+  const { doc, updateDoc, ...outerDocumentEditor } = React.useContext(
+    DocumentEditorContext
+  )
+
+  /** @type {React.ContextType<typeof DocumentEditorContext>} */
+  const documentEditor = React.useMemo(
+    () => ({
+      ...outerDocumentEditor,
+      doc,
+      updateDoc(updatedInstancePath, updatedValue) {
+        const field = updatedInstancePath.at(-1)
+        let updatedVector = set(
+          updatedInstancePath.slice(instancePath.length),
+          updatedValue
+        )(value)
+        updatedVector.version = '2.0'
+
+        updatedVector =
+          field === 'vectorString' && typeof updatedValue === 'string'
+            ? vectorUpdateFromVectorString(updatedVector)
+            : vectorUpdateVectorString(updatedVector)
+
+        updatedVector = vectorUpdateBaseScore(updatedVector)
+        updatedVector = vectorUpdateTemporalScore(updatedVector)
+        updatedVector = vectorUpdateEnvironmentalScore(updatedVector)
+        updateDoc(instancePath, updatedVector)
+      },
+    }),
+    [outerDocumentEditor, updateDoc, instancePath, value, doc]
+  )
+
+  /** @type {(childName: string, options: string[], disableClearable: boolean) => any} */
+  function dropdownFor(childName, options, disableClearable = false) {
+    const childValue = /** @type {string} */ ((value || {})[childName]) || ''
+    return cvssDropdown(
+      instancePath,
+      childName,
+      childValue,
+      options,
+      property,
+      disabled,
+      disableClearable
+    )
+  }
+
+  return (
+    <DocumentEditorContext.Provider value={documentEditor}>
+      <div className="flex flex-col gap-4 p-4 overflow-auto shrink-0 min-w-[340px]">
+        {dropdownFor('version', ['2.0'], true)}
+        <TextAttribute
+          label="VectorString"
+          description=""
+          pattern="^CVSS:3.[01]/((AV:[NALP]|AC:[LH]|PR:[UNLH]|UI:[NR]|S:[UC]|[CIA]:[NLH]|E:[XUPFH]|RL:[XOTWU]|RC:[XURC]|[CIA]R:[XLMH]|MAV:[XNALP]|MAC:[XLH]|MPR:[XUNLH]|MUI:[XNR]|MS:[XUC]|M[CIA]:[XNLH])/)*(AV:[NALP]|AC:[LH]|PR:[UNLH]|UI:[NR]|S:[UC]|[CIA]:[NLH]|E:[XUPFH]|RL:[XOTWU]|RC:[XURC]|[CIA]R:[XLMH]|MAV:[XNALP]|MAC:[XLH]|MPR:[XUNLH]|MUI:[XNR]|MS:[XUC]|M[CIA]:[XNLH])$"
```

Comment:
This is the 3.0/1 pattern, not the 2.0 one.

Reply:
that was a copy&paste error, corrected